include/deal.II/grid/grid_tools.h

```diff
@@ -983,6 +987,69 @@ namespace GridTools
   get_patch_around_cell(const typename Container::active_cell_iterator &cell);
 
 
+
+  /**
+   * This function returns a list of all parent cells of the given vector of active 
+   * cells. To get the list of cells with the coarsest common level, the function first 
+   * computes the minimum refinement level of all cells (min_level) then loops over all 
+   * cells of the given input vector (patch_cells), for each cell of the loop if its 
+   * level==min_level then the cell is pushed back to the vector of output cells,
+   * else the function looks for the parent cell of the given cell, with the level 
+   * equal to the min_level, and pushes back this parent cell to the vector of out put cells.
+   *
+   * @tparam Container
+   * In C++, the compiler can not determine the type of <code>Container</code>
+   * from the function call. You need to specify it as an explicit template
+   * argument following the function name.
+   * @param[in] patch_cells A vector of active cells.
+   * @return A list of cells with the coarsest common level of refinement
+   *
+   * @author Arezou Ghesmati, Wolfgang Bangerth, 2015
+   */
+  template<class Container >
+  std::vector<typename Container::cell_iterator> get_cells_at_coarsest_common_level(
+    const std::vector<typename Container::active_cell_iterator> &patch_cells);
+
+
+  /**
+   * The first element of input argument is a vector of active cells which we
```

Comment:
Can you try to better describe what this function does? It is typically a good idea to give a concise description by not talking about _how_ it is done, but _what_ the result is.

What are the requirements for ```patch```? Can I put in an arbitrary list of active cells? I assume there will be (potentially) other active cells that do not correspond to cells handed in via ```patch```?